R/telemetry.R

```diff
@@ -211,71 +211,7 @@ Telemetry <- R6::R6Class( # nolint object_name.
       }
 
       if (isTRUE(track_errors)) {
-        if ("onUnhandledError" %in% ls(getNamespace("shiny"))) {
-          # onUnhandledError handler is only available in shiny >= 1.8.1
-          shiny::onUnhandledError(function(error) {
-            self$log_error(
-              output_id = "global",
-              message = conditionMessage(error)
-            )
-          })
-        } else {
-          # In shiny < 1.8.1, we fallback to using the `shiny.error` option and
-          # if that is already set, we track the `shiny:error` javascript event.
-          lifecycle::deprecate_warn(
-            when = as.character(utils::packageVersion("shiny")),
-            what = "Telemetry$start_session(track_errors = \"is not fully enabled \")",
-            details = c(
-              paste(
-                "Update the shiny package to version `1.8.1` or higher to",
-                "enable logging of all errors.",
-                sep = " "
-              ),
-              paste(
-                "Until then, shiny.telemetry can only reliably detect errors",
-                "triggered by the `shiny:error` javacript event.",
-                sep = " "
-              )
-            ),
-            env = getNamespace("shiny")
-          )
-
-          if (is.null(getOption("shiny.error"))) {
-            options(
-              "shiny.error" = function(.envir = parent.frame()) {
-                # make sure id is a string without spaces
-                output_id <- paste(
-                  gsub(
-                    " |\t|\r",
-                    "_",
-                    as.character(.envir$e$call %||% "global")
-                  ),
-                  collapse = "__"
-                )
-
-                self$log_error(
-                  output_id = output_id,
-                  message = .envir$e$message %||% "Unknown error.",
-                  session = session
-                )
-              }
-            )
-
-            # Restore previous option
-            shiny::onSessionEnded(
-              fun = function() options("shiny.error" = NULL),
-              session = session
-            )
-          } else {
-            shiny::observeEvent(input[[private$.track_error_id]], {
-              self$log_error(
-                output_id = input[[private$.track_error_id]]$output_id,
-                message = input[[private$.track_error_id]]$message,
-                session = session
-              )
-            })
-          }
-        }
+        self$log_errors(session)
```

Comment:
Extract logic away from `start_session` to its own function.

Reply:
I think it is good move to take the logic out of `start_session`.
Does this mean the user can now use the `log_errors` function outside of `start_session`?

Reply:
Yup, they can if they wish, just like the rest of the API to bring consistency.

It will generate duplicate calls when called along `start_session`, but that is user preference